batchflow/utils_transforms.py

```diff
@@ -0,0 +1,217 @@
+                array += min_
+
+        return array
+
+class Quantizer:
+    """ Class to hold parameters and methods for (de)quantization.
+
+    Parameters
+    ----------
+    ranges : tuple
+        Bounds to create bins.
+    clip : bool, optional
+        Whether to clip data to selected ranges, by default True
+    center : bool, optional
+        Whether to make data have 0-mean before quantization, by default False
+    mean : _type_, optional
+        Mean value for centering, by default None
+    dtype : numpy.dtype, optional
+        dtype for the quantized array, by default np.int8
+    copy : bool, optional
+        Whether to make copy of the data under the hood, by default False.
+        Enabled copy will not allow to change input data but quantization will be slower.
+    """
+    def __init__(self, ranges, clip=True, center=False, mean=None, dtype=np.int8, copy=False):
+        if dtype not in [np.int8, np.int16, np.uint8, np.uint16]:
+            raise TypeError(f'{dtype} is not supported, use int8, int16, uint8, uint16')
+        self.ranges = ranges
+        self.clip, self.center = clip, center
+        self.mean = mean
+        self.dtype = dtype
+        self.copy = copy
+
+        n_bins = np.iinfo(dtype).max - np.iinfo(dtype).min - 1
+        self.bins = np.histogram_bin_edges(None, bins=n_bins, range=ranges).astype(np.float32)
+
+    def quantize(self, array):
+        """ Quantize data: find the index of each element in the pre-computed bins and use it as the new value.
+        Converts `array` to `self.dtype`. Lossy.
+        """
+        if self.copy:
+            array = array.copy()
+        if self.center:
+            array -= self.mean
+        if self.clip:
+            array = np.clip(array, *self.ranges)
+
+        array = np.digitize(array, self.bins, right=False) + np.iinfo(self.dtype).min
+        if not self.clip:
+            array[array == np.iinfo(self.dtype).max + 1] = np.iinfo(self.dtype).max # to put maximum value into bin
+        return array.astype(self.dtype)
+
+    def dequantize(self, array):
+        """ Dequantize data: use each element as the index in the array of pre-computed bins.
+        Converts `array` to float32 dtype. Unable to recover full information.
+        """
+        if self.copy:
+            array = array.copy()
+        array = array.astype(np.int32) - np.iinfo(self.dtype).min
+        np.clip(array, 0, len(self.bins)-1, out=array)
+        array = self.bins[array]
+        if self.center:
+            array += self.mean
+        return array.astype(np.float32)
+
+    def compute_mean_error(self, data):
+        """ Estimate quantization error on data. """
+        quantized_data = self.quantize(data)
+        dequantized_data = self.dequantize(quantized_data)
+        return np.mean(np.abs(dequantized_data - data)) / data.std()
+
+    @property
+    def estimated_absolute_error(self):
+        return np.diff(self.bins).max()
+
+    def __call__(self, array):
```

Comment:
Put it in the same relative position, as in `Normalizer`: right after the `normalize` method.

Also, you can write `__call__ = quantize` and that would save the documentation / signature / etc